### PR TITLE
Server tests. #288

### PR DESCRIPTION
**Pull request by romanstetsyk**
# Description

Added Jest tests for Express middleware:
To run tests: `npm run test`

## Type of change

Please select everything applicable. Please, do not delete any lines.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Issue
- [x] Is this related to a specific issue? Is so please specify: #288 and #302

# Checklist:

- [x] This PR is up to date with the development branch, and merge conflicts have been resolved
- [x] I have executed `npm run lint` and resolved any outstanding errors. Most issues can be solved by executing `npm run format`
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: server/test/unit/createEventsArray.test.js [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+  groupId: null,
+  rsvp: [],
+};
+
+const nonRecurring = [
+  {
+    description: "starts and ends on the same UTC day",
+    input: {
+      ...commonFormFields,
+      firstEventStart: Date.parse("2023-01-28T15:00:00.000+00:00"),
```
Comment (MatthewBozin) [inline server/test/unit/createEventsArray.test.js:33]: I received typeerrors in all the tests in this sequence when I tried to run it! It looks like we're trying to feed the createEventsArray function an object with the properties "firstEventStart", "firstEventEnd", and "lastEventStart" when it's expecting to receive "initialDate", "finalDate", "startTime", and "endTime".

**Comment by MatthewBozin** [comment]
I went through and tried out the test suite! Everything worked except for createEventsArray (left a comment regarding this).

File: server/routes/events.js [hunk 2]
```diff
@@ -10,8 +10,8 @@ const { createEventSchema } = require("../models/Event");
 router.post(
   "/",
   auth.ensureAuth,
-  validateBody(createEventSchema),
-  maxEvents,
+  // validateBody(createEventSchema),
```
Comment (MatthewBozin) [inline server/routes/events.js:13]: It looks like when I uncomment these, creating an event returns a 400 error.

File: client/src/features/form/FormMoverControl.js [hunk 3]
```diff
@@ -78,9 +78,9 @@ const FormMoverControl = () => {
         // Compares the date + time combined (i.e. start and end datetime)
         const startDateTime = new Date(formData["initialDate"] + " " + formData["startTime"]);
         const endDateTime = new Date(formData["finalDate"] + " " + formData["endTime"]);
-        if (endDateTime < startDateTime) {
-          errorArray.push("Error: End time is before Start time");
-        }
+        // if (endDateTime < startDateTime) {
```
Comment (MatthewBozin) [inline client/src/features/form/FormMoverControl.js:81]: Do we want this to remain commented out?

**Review by MatthewBozin — CHANGES_REQUESTED** [review]
Hey Roman! 
I tried out the test suite! All test files worked fine except for createEventsArray (left a comment regarding the likely cause). While the app works fine, I noticed you've commented out a few of the form validation checks on the client and server side and wanted to know more about that before we merge this PR.
Thanks!
-Matt

**Comment by romanstetsyk** [comment]
@MatthewBozin thank you for reviewing this PR. This is a work in progress. It was made before we discovered bugs dealing with time zones and daylight-saving time. I am considering different approaches to solving this, and the tests haven't been updated yet.

**Comment by MatthewBozin** [comment]
Ah gotcha! Thanks for letting me know!

**Comment by romanstetsyk** [comment]
@MatthewBozin I've fixed the timezone bug (hopefully). Could you review the updated PR? We might need to add a warning when someone tries to create an event when time switches from normal to daylight-saving, as the display times may be misleading. But the most important thing is that in the database, the UTC times are saved correctly!

**Comment by Caleb-Cohen** [comment]
Added [a commit](https://github.com/Together-100Devs/Together/pull/289/commits/1eb314e68f9a66aa40b35d86d8e73e35ef2042b9) please review.

**Pull request merged**